src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -260,7 +260,12 @@ public OptionSet Options
 
         internal void UpdateCurrentSolutionOnOptionsChanged()
         {
-            SetCurrentSolution(CurrentSolution.WithOptions(new SolutionOptionSet(_legacyOptions)));
+            using (_serializationLock.DisposableWait())
```

Comment:
SetCurrentSolution is locked in all other places in Workspace (except for two methods).  Seems highly likely they need to follow the rest here.